llvm/lib/Object/Minidump.cpp

```diff
@@ -53,6 +53,27 @@ Expected<std::string> MinidumpFile::getString(size_t Offset) const {
   return Result;
 }
 
+Expected<std::vector<const minidump::ExceptionStream *>>
+MinidumpFile::getExceptionStreams() const {
```

Comment:
This looks like a job for an (non-fallible) input iterator returning an `Expected<const ExceptionStream&>`. (non-fallible, because the iteration always suceeds, but reading of any stream can fail, and can do so independendly of all other streams.)

Alternatively, maybe we could just return an array of minidump::Directory descriptors, and have the caller retrieve them one at a time via something like `getExceptionStream(Directory)`.